src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -137,59 +190,83 @@ public Task<RestoreResult> ExecuteAsync()
+                        msbuildOutputFiles = BuildAssetsUtils.GetMSBuildOutputFiles(
+                            _request.Project,
+                            assetsFile,
+                            graphs,
+                            localRepositories,
+                            _request,
+                            assetsFilePath,
+                            _success,
+                            _logger);
+                    }
 
-                if (contextForProject.IsMsBuildBased)
-                {
-                    msbuildOutputFiles = BuildAssetsUtils.GetMSBuildOutputFiles(
-                        _request.Project,
-                        assetsFile,
-                        graphs,
-                        localRepositories,
-                        _request,
-                        assetsFilePath,
-                        _success,
-                        _logger);
-                }
+                    // If the request is for a lower lock file version, downgrade it appropriately
+                    DowngradeLockFileIfNeeded(assetsFile);
 
-                // If the request is for a lower lock file version, downgrade it appropriately
-                DowngradeLockFileIfNeeded(assetsFile);
+                    // Revert to the original case if needed
+                    await FixCaseForLegacyReaders(graphs, assetsFile, token);
 
-                // Revert to the original case if needed
-                await FixCaseForLegacyReaders(graphs, assetsFile, token);
+                    // Write the logs into the assets file
+                    var logs = _logger.Errors
+                        .Select(l => AssetsLogMessage.Create(l))
+                        .ToList();
 
-                // Write the logs into the assets file
-                var logs = _logger.Errors
-                    .Select(l => AssetsLogMessage.Create(l))
-                    .ToList();
+                    _success &= !logs.Any(l => l.Level == LogLevel.Error);
 
-                _success &= !logs.Any(l => l.Level == LogLevel.Error);
+                    assetsFile.LogMessages = logs;
 
-                assetsFile.LogMessages = logs;
+                    if (cacheFile != null)
+                    {
+                        cacheFile.Success = _success;
+                    }
 
-                if (cacheFile != null)
-                {
-                    cacheFile.Success = _success;
+                    var errorCodes = ConcatAsString(logs.Where(l => l.Level == LogLevel.Error).Select(l => l.Code));
+                    var errorMessages = ConcatAsString(logs.Where(l => l.Level == LogLevel.Error).Select(l => l.Message));
+                    var warningCodes = ConcatAsString(logs.Where(l => l.Level == LogLevel.Warning).Select(l => l.Code));
+                    var warningMessages = ConcatAsString(logs.Where(l => l.Level == LogLevel.Warning).Select(l => l.Message));
+
+                    if (!string.IsNullOrEmpty(errorCodes))
```

Comment:
I think VS Data API has some specified TelemetryEvent for Error/warnings, Could we look at those API to see if there is any benefit for us? we all used most basic telemetryEvent API here, we might can consider more specified API.